src/encoded/types/dataset.py

```diff
@@ -1008,6 +1046,37 @@ def biosample_summary(self, request, related_datasets):
         if all_ontologies and not all_summaries:
             return all_terms
 
+    @calculated_property(schema={
+        "title": "Files",
+        "type": "array",
+        "items": {
+            "type": "string",
+            "linkTo": "File",
+        },
+    })
+    def series_files(self, request, original_files, related_datasets, status):
+        related_datasets_paths = paths_filtered_by_status(request, related_datasets)
+        original_related_datasets_files = []
+        for path in related_datasets_paths:
+            related_dataset = request.embed(
+                path,
+                '@@object_with_select_calculated_properties'
+                '?field=@id'
+                '&field=original_files'
+            )
+            original_related_datasets_files.extend(related_dataset.get('original_files', []))
```

Comment:
We don't have to worry about the same file showing up in multiple `original_files` right?

Reply:
I think it’s possible. The cart code deduplicates files. Not sure on the file gallery.